base_layer/wallet/src/transaction_service/config.rs

```diff
@@ -52,32 +53,48 @@ impl Default for TransactionServiceConfig {
             pending_transaction_cancellation_timeout: Duration::from_secs(259200), // 3 Days
             num_confirmations_required: 3,
             peer_dial_retry_timeout: Duration::from_secs(20),
+            transaction_sending_mechanism: TransactionSendingMechanism::default(),
         }
     }
 }
 
-impl TransactionServiceConfig {
-    pub fn new(
-        broadcast_monitoring_timeout: Duration,
-        chain_monitoring_timeout: Duration,
-        direct_send_timeout: Duration,
-        broadcast_send_timeout: Duration,
-    ) -> Self
-    {
-        trace!(
-            target: LOG_TARGET,
-            "Timeouts - Broadcast monitoring: {} s, Chain monitoring: {} s, Direct send: {} s, Broadcast send: {} s",
-            broadcast_monitoring_timeout.as_secs(),
-            chain_monitoring_timeout.as_secs(),
-            direct_send_timeout.as_secs(),
-            broadcast_send_timeout.as_secs(),
-        );
-        Self {
-            broadcast_monitoring_timeout,
-            chain_monitoring_timeout,
-            direct_send_timeout,
-            broadcast_send_timeout,
-            ..Default::default()
+#[derive(Clone, PartialEq, Debug)]
```

Comment:
I have realized that you should add Copy to these types of Enums. You still need clone to play nice with the larger Clonable structs this lives inside of but if you add Copy here you don't need to explicitly clone these anymore, like you do further down.